File: Server/src/main/java/com/daffodil/renters/core/model/beans/Listing.java
package com.daffodil.renters.core.model.beans;

import com.daffodil.renters.core.model.beans.postables.Amenities;
import com.daffodil.renters.core.model.beans.postables.ParkingSpot;
import com.daffodil.renters.core.model.beans.postables.Seller;
import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Getter;
import lombok.Setter;

import java.util.Date;
import java.util.List;

@JsonInclude(JsonInclude.Include.NON_EMPTY)
public class Listing extends ListingSkeletal {

    @Getter
    @Setter
    String description;
    @Getter
    @Setter
    Amenities amenities;
    @Getter
    @Setter
    Seller seller;
    @Getter
    @Setter
    long securityDeposit;
    @Getter
    @Setter
    long brokerage;
    @Getter
    @Setter
    int ageOfPropertyMonths;
    @Getter
    @Setter
    int lockInPeriod;
    @Getter
    @Setter
    Date listedOn;
    @Getter
    @Setter
    List<ParkingSpot> sharedParkingSpots;

    public static class Filter extends ListingSkeletal.Filter {

        public Filter() {
        }
    }

}

File: Android-Client/app/src/main/java/com/daffodil/renters/model/Listing.kt
package com.daffodil.renters.model

import com.daffodil.renters.model.postables.Amenities
import com.daffodil.renters.model.postables.ParkingSpot
import com.daffodil.renters.model.postables.Seller
import java.util.*

class Listing : ListingSkeletal() {

    var description: String? = null
    var amenities: Amenities? = null
    var seller: Seller? = null
    var securityDeposit: Long = 0
    var brokerage: Long = 0
    var ageOfPropertyMonths: Int = 0
    var lockInPeriod: Int = 0
    var listedOn: Date? = null
    var sharedParkingSpots: List<ParkingSpot>? = null

    inner class FormattedStrings : ListingSkeletal.FormattedStrings() {

        fun getSellerFullNameSentenceCase() =
            "${seller?.firstName?.toLowerCase()?.capitalize()} ${seller?.lastName?.toLowerCase()?.capitalize()}"

        fun getSellerTypeSentenceCase() =
            seller?.sellerType?.toString()?.toLowerCase()?.capitalize()

    }

}
